src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs

```diff
@@ -166,7 +167,8 @@ public IUrlHelper Url
             {
                 if (_url == null)
                 {
-                    _url = Resolver?.GetRequiredService<IUrlHelper>();
+                    var factory = Resolver.GetRequiredService<IUrlHelperFactory>();
+                    _url = factory?.GetUrlHelper(ControllerContext);
```

Comment:
You nuked the null checks on all other locations like this. Why's this one different?